app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+
+    [Person(person["name"], person["age"]) for person in people]
+
+    for person in people:
+        person_instance = Person.people[person["name"]]
+
+        if "wife" in person and person.get("wife"):
```

Comment:
just to make more beautiful you can use this:
```suggestion
        if wife := person.get("wife"):
```